lib/vrl/parser/src/arbitrary.rs

```diff
@@ -0,0 +1,243 @@
+//! Arbitrary instances for the AST nodes.
+//! This allows us to generate random VRL nodes for the purposes of fuzz testing.
+//! We put as little effort as is reasonable to generate valid programs. The nodes are passed to the
+//! compiler which rejects invalid scripts. This prevents us from imposing too much structure on the
+//! code that builds in assumptions we have made about the code which gives the fuzzer more freedom
+//! to find areas of nodes that we haven't thought of.
+use crate::{
+    arbitrary_depth::ArbitraryDepth,
+    ast::{
+        Assignment, AssignmentOp, AssignmentTarget, Block, Container, Expr, FunctionArgument,
+        FunctionCall, Group, Ident, IfStatement, Node, Op, Opcode, Predicate, Query, QueryTarget,
+        RootExpr,
+    },
+    Literal, Program,
+};
+use arbitrary::{Arbitrary, Unstructured};
+use diagnostic::Span;
+use lookup::LookupBuf;
+
+const DEPTH: isize = 4;
```

Comment:
Fuzzing turn up anything interesting if you increase this, or does increasing this blow up the run?

Reply:
The bigger the number, it seems the odds of generating a valid VRL program rapidly approach zero - so no tests get run.

Reply:
Heard, makes sense.